src/core/vdom/helpers/normalize-children.js

```diff
@@ -45,7 +45,11 @@ function normalizeArrayChildren (children: any, nestedIndex?: string): Array<VNo
     last = res[res.length - 1]
     //  nested
     if (Array.isArray(c)) {
-      res.push.apply(res, normalizeArrayChildren(c, `${nestedIndex || ''}_${i}`))
```

Comment:
I think here only need to determine whether `c` is created by `slot`, avoid to set `key ` twice